### ISSUE TITLE
MDTextField not validating on text change

### ISSUE THREAD

**Issue by BeardyBarber** [kivymd/KivyMD#851]
### Description of the Bug

MDTextField is not validating on input only when it lose or gain focus.

Reproduction steps for below code:
1. Run example (text field gain focus on start) 
2. Write something (helper text still visible)
3. Lose focus clicking on background (helper text disappear)

In my understanding when typing validation should be made after any change to text but now it is only made on focus in and out.

### Code and Logs

```from kivy.lang import Builder

from kivymd.app import MDApp

KV = '''
MDTextField:
    hint_text: "required = True"
    required: True
    helper_text_mode: "on_error"
    helper_text: "Enter text"
    focus: True
    pos_hint: {'center_x': .5, 'center_y':.5}
'''


class TestCard(MDApp):
    def build(self):
        return Builder.load_string(KV)


TestCard().run()

```

### Screenshots
![Peek 2021-02-14 13-03](https://user-images.githubusercontent.com/20884974/107883047-c7e5de80-6eec-11eb-9709-3fe8b766c61f.gif)

### Versions

* OS: MINT 20.1
* Python: 3.8.5
* Kivy: master (2021-02-14)
* KivyMD: master (2021-02-13)

**Comment by HeaTTheatR** [comment]
@BeardyBarber Please describe your question in one message, attach a minimal code that may be run without additional manipulations, so developers can look at the working script and fix it. Also write what you expect from the script and what happened wrong. If needed, attach links or screenshots to explain us what you want to do.

**Comment by BeardyBarber** [comment]
@HeaTTheatR I'm sorry, I edited my issue. Probably accidentally refreshed site after I filled the whole issue and didn't noticed it get cleared...

[Edit] Also the `error` attribute for MDTextField always return False even if it is empty, let me know if it needs another issue.

**Comment by podraco** [comment]
MDTextField is based in kivy.uix.textinput, there it allows you to see what events are dispatched every time you type something inside.

**Comment by BeardyBarber** [comment]
@podraco Can you tell me how can i get to these events and use this knowledge to my needs?

**Comment by runette** [comment]
As far as I understand the issue - the fix for #858 contained in PR #859 also fixes this issue - see the following 

![Feb-15-2021 22-23-02](https://user-images.githubusercontent.com/2239795/107998228-8cc2d880-6fdc-11eb-97db-8ec32c640c35.gif)

**Comment by BeardyBarber** [comment]
Working as expected, closing.
